### ISSUE TITLE
issue about prediction

### ISSUE THREAD

**Issue by zhyluo** [tensorflow/ranking#183]
I train_and_evaled the model like this: tf.estimator.train_and_evaluate(rank_estimator, train_spec, vali_spec) and got a result  metric/ordered_pair_accuracy = 1.0 on the validation set.
But, the problem seems that when I ran rank_estimator.predict(input_fn=vali_input_fn) alone on the same validation set based on the saved best model, I got a totally  unordered list by predicted score.
Am I making a mistake here? thx!

btw:
I used the same vali_input_fn in train_and_eval and predict as follows:
    def _eval_input_fn():
        """Defines training input fn."""
        file_paths = [fname]
        dataset = tf.data \
            .TFRecordDataset(file_paths) \
            .map(lambda x: reader(config, x), num_parallel_calls=tf.data.experimental.AUTOTUNE) \
            .repeat(1) \
            .batch(config.batch_size).prefetch(buffer_size=tf.data.experimental.AUTOTUNE)

**Comment by pluspluswu** [comment]
mark
trying rewrite model_fn to do prediction here

**Comment by xuanhuiwang** [comment]
Do you mean the predicted scores are not sorted? Or do you mean the list sorted by the predicted scores are not perfect to have the ordered_pair_accuracy = 1.0?
